File: v23_command/python/light_controller1.py
from abc import ABC, abstractmethod

class Light(ABC):
    @abstractmethod
    def on(self):
        pass

    @abstractmethod
    def off(self):
        pass

class LivingRoomLight(Light):
    def on(self):
        print("Living room light is on")

    def off(self):
        print("Living room light is off")

class KitchenLight(Light):
    def on(self):
        print("Kitchen light is on")

    def off(self):
        print("Kitchen light is off")

class LightController1:
    def main(self):
        living_room_light = LivingRoomLight()
        kitchen_light = KitchenLight()

        kitchen_light.on()
        kitchen_light.off()
        living_room_light.on()
        living_room_light.off()

if __name__ == "__main__":
    light_controller1 = LightController1()
    light_controller1.main()

File: v23_command/java/LightController1.java
package v23_command.java;

interface Light {
    void on();
    void off();
}

class LivingRoomLight implements Light {
    public void on() {
        System.out.println("Living room light is on");
    }

    public void off() {
        System.out.println("Living room light is off");
    }
}

class KitchenLight implements Light {
    public void on() {
        System.out.println("Kitchen light is on");
    }

    public void off() {
        System.out.println("Kitchen light is off");
    }
}

public class LightController1 {
    public static void main(String[] args) {
        LivingRoomLight livingRoomLight = new LivingRoomLight();
        KitchenLight kitchenLight = new KitchenLight();

        kitchenLight.on();
        kitchenLight.off();
        livingRoomLight.on();
        livingRoomLight.off();
    }
}